pallets/providers/src/utils.rs

```diff
@@ -1135,6 +1249,44 @@ where
 
         Ok(())
     }
+
+    /// Compute the deposit needed for a given capacity.
+    pub(crate) fn compute_deposit_needed_for_capacity(
+        capacity: T::StorageDataUnit,
+    ) -> Result<BalanceOf<T>, DispatchError> {
+        let capacity_over_minimum = capacity
+            .checked_sub(&T::SpMinCapacity::get())
+            .ok_or(Error::<T>::StorageTooLow)?;
+        let deposit_for_capacity_over_minimum = T::DepositPerData::get()
+            .checked_mul(&capacity_over_minimum.into())
+            .ok_or(ArithmeticError::Overflow)?;
+        T::SpMinDeposit::get()
+            .checked_add(&deposit_for_capacity_over_minimum)
+            .ok_or(ArithmeticError::Overflow.into())
+    }
+
+    fn get_provider_details(
+        provider_id: ProviderIdFor<T>,
+    ) -> Result<(T::AccountId, StorageDataUnit<T>, StorageDataUnit<T>), DispatchError>
```

Comment:
Not a huge fan of returning a tuple here, but ok.

Reply:
ya it's just a convenience function for extracting the exact values I need for the slashing and top up operations. I don't want to create another struct or return an enum just for this.